yaml_test.go

```diff
@@ -118,6 +118,87 @@ func unmarshal(t *testing.T, y []byte, s, e interface{}, opts ...JSONOpt) {
 	}
 }
 
+func TestUnmarshalStrict(t *testing.T) {
+	y := []byte("a: 1")
+	s1 := UnmarshalString{}
+	e1 := UnmarshalString{A: "1"}
+	unmarshalStrict(t, y, &s1, &e1)
+
+	y = []byte("a: true")
```

Comment:
these should have comments on what each individual test is, or better yet be table tests using `t.Run()`

Reply:
@liztio this is patterned on existing tests above :)